src/Container/InMemoryContainerLog.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Container;
+
+use Exception;
+use Tempest\Container\Exceptions\CircularDependencyException;
+
+final class InMemoryContainerLog implements ContainerLog
+{
+    private string $origin = '';
+
+    public function __construct(
+        /** @var \Tempest\Container\Context[] $stack */
+        private array $stack = [],
+
+        /** @var \Tempest\Container\Dependency[] $stack */
+        private array $dependencies = [],
```

Comment:
Is there any real reason to have these exposed?

Reply:
No, this is a remnant of my first implementation where I did circular checks on dependencies instead of contexts, which didn't make any sense. I'll remove it